Before: 4f525208ba899c1df2ee4f64a9fefbfd01586700fc1c071a7b7b80abbe44e29f
After: aa5716e985baeb09120aba9371889e32e26d261c1d28850ce5963f7f98149be4
Patch: 3d2a93c0a904539996e86124fd8023ae0a1c6648162ea2de0bbcef0e0803f865
--- /dev/null	1980-01-01 00:00:00.000000000 +1000
+++ .corvid/Gemfile	2012-09-06 07:57:58.000000000 +1000
@@ -0,0 +1,30 @@
+# vim:ft=ruby et ts=2 sw=2:
+
+gem 'corvid'
+
+gem 'rake', group: :rake
+
+group :test do
+  gem 'ci_reporter', require: false
+  gem 'simplecov', '>= 0.6.4', require: false
+end if Dir.exists?('test')
+
+group :test_unit do
+  gem 'minitest'
+  gem 'guard-minitest', '>= 0.5.0', require: false
+  gem 'turn', '>= 0.9.4'
+end if Dir.exists?('test/unit')
+
+group :test_spec do
+  gem 'rspec'
+  gem 'guard-rspec', require: false
+end if Dir.exists?('test/spec')
+
+group :doc do
+  gem 'yard'
+end
+
+group :debug do
+  gem 'debugger', platforms: :mri
+end
+
--- /dev/null	1980-01-01 00:00:00.000000000 +1000
+++ Gemfile	2012-09-06 07:57:58.000000000 +1000
@@ -0,0 +1,12 @@
+source :rubygems
+gemspec
+
+# Load Corvid dependencies
+eval File.read(File.expand_path '../.corvid/Gemfile', __FILE__)
+
+# Parser for Markdown documentation
+group :doc do
+  gem 'rdiscount', platforms: :mri
+  gem 'kramdown', platforms: :jruby
+end
+
--- Gemfile.tt	2012-09-20 22:00:04.000000000 +1000
+++ /dev/null	1980-01-01 00:00:00.000000000 +1000
@@ -1,39 +0,0 @@
-source :rubygems
-gemspec
-
-gem 'corvid'
-
-gem 'rake', group: :rake
-
-<% if any_features_installed? :test_unit, :test_spec -%>
-group :test do
-  gem 'ci_reporter', require: false
-  gem 'simplecov', '>= 0.6.4', require: false
-  gem 'guard', '>= 1.3.2', require: false
-  gem 'rb-inotify', require: false
-end
-
-<% end -%>
-<% if feature_installed? :test_unit -%>
-group :test_unit do
-  gem 'minitest'
-  gem 'guard-minitest', '>= 0.5.0', require: false
-  gem 'turn', '>= 0.9.4'
-end
-
-<% end -%>
-<% if feature_installed? :test_spec -%>
-group :test_spec do
-  gem 'rspec'
-  gem 'guard-rspec', require: false
-end
-
-<% end -%>
-group :doc do
-  gem 'yard'
-  # Parsers for Markdown documentation
-  gem 'rdiscount', platforms: :mri
-  gem 'kramdown', platforms: :jruby
-end
-
-# vim:ft=ruby et ts=2 sw=2:
--- corvid-features/corvid.rb	2012-09-20 21:04:13.000000000 +1000
+++ corvid-features/corvid.rb	2012-09-13 10:10:04.000000000 +1000
@@ -6,17 +6,18 @@
   copy_executable 'bin/yardoc'
   copy_executable 'bin/yri'
 
+  copy_file       '.corvid/Gemfile'
   copy_file       '.corvid/stats_cfg.rb'
   copy_file       '.corvid/todo_cfg.rb'
   copy_file       '.gitignore'
   copy_file       '.simplecov'
   copy_file       '.yardopts'
   copy_file       'CHANGELOG.md'
+  copy_file       'Gemfile'
   copy_file       'Guardfile'
   copy_file       'Rakefile'
   copy_file       'README.md'
 
-  template2       'Gemfile.tt'
   template2       'lib/%project_name%/version.rb.tt'
   template2       '%project_name%.gemspec.tt'
 }
--- corvid-features/test_spec.rb	2012-09-20 21:04:13.000000000 +1000
+++ corvid-features/test_spec.rb	2012-09-06 07:57:58.000000000 +1000
@@ -1,11 +1,10 @@
 requirements 'corvid:corvid'
 
 install {
-  copy_file                        '.rspec'
-  copy_executable                  'bin/guard'
-  copy_executable                  'bin/rspec'
-  copy_file_unless_exists          'test/bootstrap/all.rb'
-  copy_file                        'test/bootstrap/spec.rb'
-  empty_directory                  'test/spec'
-  regenerate_template_with_feature 'Gemfile.tt'
+  copy_file               '.rspec'
+  copy_executable         'bin/guard'
+  copy_executable         'bin/rspec'
+  copy_file_unless_exists 'test/bootstrap/all.rb'
+  copy_file               'test/bootstrap/spec.rb'
+  empty_directory         'test/spec'
 }
--- corvid-features/test_unit.rb	2012-09-20 21:04:13.000000000 +1000
+++ corvid-features/test_unit.rb	2012-09-06 07:57:58.000000000 +1000
@@ -1,9 +1,8 @@
 requirements 'corvid:corvid'
 
 install {
-  copy_executable                  'bin/guard'
-  copy_file_unless_exists          'test/bootstrap/all.rb'
-  copy_file                        'test/bootstrap/unit.rb'
-  empty_directory                  'test/unit'
-  regenerate_template_with_feature 'Gemfile.tt'
+  copy_executable         'bin/guard'
+  copy_file_unless_exists 'test/bootstrap/all.rb'
+  copy_file               'test/bootstrap/unit.rb'
+  empty_directory         'test/unit'
 }
